src/hooks/useCheckVoucherValidity/useCheckVoucherValidity.tsx

```diff
@@ -0,0 +1,95 @@
+import { useState, useCallback } from "react";
+import { validateVoucherCode } from "../../utils/validateVoucherCode";
+import { getQuota } from "../../services/quota";
+import { Quota, Voucher } from "../../types";
+
+export class ScannerError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "Error Scanning";
+  }
+}
+
+export class InvalidVoucherError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "InvalidVocuherError";
+  }
+}
+
+type CheckValidityState = "DEFAULT" | "CHECKING_VALIDITY" | "RESULT_RETURNED";
+
+export type useCheckVoucherValidity = {
+  checkValidityState: CheckValidityState;
+  checkValidity: (serial: string, vouchers: Voucher[]) => void;
+  validityResult?: Voucher;
+  error?: Error;
+  resetValidityState: () => void;
+};
+
+export const useCheckVoucherValidity = (
+  authKey: string,
+  endpoint: string
+): useCheckVoucherValidity => {
+  const [checkValidityState, setCheckValidityState] = useState<
+    CheckValidityState
+  >("DEFAULT");
+  const [validityResult, setValidityResult] = useState<Voucher>();
+  const [error, setError] = useState<Error>();
+
+  const resetValidityState = useCallback((): void => {
+    setError(undefined);
+    setCheckValidityState("DEFAULT");
+  }, []);
+
+  const getVoucherValidity = (response: Quota, serial: string): Voucher => {
+    const voucherQuota = response.remainingQuota.filter(
+      quota => quota.category === "voucher"
+    )[0];
+    if (voucherQuota.quantity === 0)
+      throw new InvalidVoucherError("Item redeemed previously");
+    return {
+      serial,
+      denomination: 2
+    };
+  };
+
+  const checkValidity: useCheckVoucherValidity["checkValidity"] = useCallback(
+    async (serial, vouchers): Promise<void> => {
+      setCheckValidityState("CHECKING_VALIDITY");
+      const check = async (): Promise<void> => {
+        if (serial.length === 0) {
+          setError(new Error("No voucher code entered"));
+          return;
+        }
+
+        try {
+          const serialArr = vouchers.map(voucher => voucher.serial);
+          validateVoucherCode(serial, serialArr);
+        } catch (e) {
+          setError(new ScannerError(e.message));
+          return;
+        }
+        //Send to backend to check if valid
+        try {
+          const validityResponse = await getQuota([serial], authKey, endpoint);
+          const voucher = getVoucherValidity(validityResponse, serial);
+          setValidityResult(voucher);
+          setCheckValidityState("RESULT_RETURNED");
+        } catch (e) {
+          setError(e);
```

Comment:
Note that we'll need to handle the case when the voucher isn't in the whitelist. Will let you guys know again.